packages/perseus/src/widgets/cs-program/cs-program.tsx

```diff
@@ -64,25 +62,51 @@ class CSProgram extends React.Component<Props> implements Widget {
     };
 
     componentDidMount() {
-        $(window).on("message", this.handleMessageEvent);
+        window.addEventListener("message", this.handleMessageEvent);
     }
 
     componentWillUnmount() {
-        $(window).off("message", this.handleMessageEvent);
+        window.removeEventListener("message", this.handleMessageEvent);
     }
 
-    handleMessageEvent: (arg1: any) => void = (e) => {
+    handleMessageEvent = (e: MessageEvent): void => {
         // We receive data from the iframe that contains {passed: true/false}
         //  and use that to set the status
+        // Note: Origin might be null or "" in file:// or test environments.
+        const allowedOrigins = [
+            "https://khanacademy.org",
+            "https://www.khanacademy.org",
+            "https://ka.org",
+            "https://www.ka.org",
+            // Add specific testing origins if needed, e.g., "http://localhost:xxxx"
+        ];
+
```

Comment:
Not sure about these security measures. This is for all usages, and I know we have changes coming up for urls, so leaning towards removing this and some of the measures below.

Reply:
Had to modify your suggestion a bit to get it to work, but I think it's working. Had to manually add a source to the tests and it looks like to match the MessageEventSource type, we had to reference the contentWindow. Also added iframeElement as a property on CSProgram; wasn't sure if that's what was intended here or if you were trying to access some inherent property that should already be there.